### ISSUE TITLE
undefined context._shadow" error

### ISSUE THREAD

**Issue by tigeugene** [phloose/leaflet-layervisibility#1]
I am trying to use your leaflet plugin but am getting an "undefined context._shadow" error(details below) when trying to hide or show a leaflet Marker.

Cannot figure out what I am doing wrong!

I am using Leaflet v1.7.1
Browsers tested : Microsoft Edge(Version 98.0.1108.43) and Firefox(v96.0.3)

  
  <script src="lib/leaflet/leaflet.js"></script>
  <link rel="stylesheet" href="lib/leaflet/leaflet.css" />
  <script src="https://unpkg.com/leaflet-layervisibility/dist/leaflet-layervisibility.js"></script>
 
  // JAVASCRIPT code
  eventLocationMarker = L.marker([locationLong, locationLat]);
  eventLocationMarker.hide();

Browser console stacktrace:  
Uncaught TypeError: context._shadow is undefined
    setMarkerShadowDisplayStyle leaflet-layervisibility.js:51
    hide leaflet-layervisibility.js:57
    <anonymous> map_page.js:398
    leaflet-layervisibility.js:51:4
    setMarkerShadowDisplayStyle leaflet-layervisibility.js:51
    hide leaflet-layervisibility.js:57
    <anonymous> map_page.js:398

**Comment by phloose** [comment]
Hey, I will have a look into this! Did you already add the markers to a map instance?

**Comment by tigeugene** [comment]
oops - just figured this out.
My app is not adding the marker directly to the map using addTo(map).
It is adding it to a clustered Group using the leaflet plugin. https://github.com/Leaflet/Leaflet.markercluster which works very well.
When I also add ".addTo(map);" to the line where I create the Marker then this plugin also works well.

**Comment by phloose** [comment]
Interesting 🤔 maybe it has something to do with how the cluster plugin adds content to a map. Will keep you posted

**Comment by ericek111** [comment]
This happens when you try to hide a marker with a `DivIcon`. Such marker does not have the `_shadow` property. Simply checking it before makes it work.

```
    function setMarkerShadowDisplayStyle(value, context) {
        if (!context._shadow)
            return;
         ....
```

**Comment by malcomio** [comment]
I've encountered similar problems, as mentioned in #4 

as per the comment from @ericek111 it is for markers with icons, added as geoJson

seems to be addressed by #6
